Polio 2034 performance dashboard threshold setting

Crebert08 opened the pull request:
What problem is this PR solving? Explain here in one sentence.


Related JIRA tickets : [POLIO-2034](https://bluesquare.atlassian.net/browse/POLIO-2034)

## Self proofreading checklist

- [x] Did I use eslint and ruff formatters?
- [x] Is my code clear enough and well documented?
- [ ] Are my typescript files well typed?
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [x] Are there enough tests?
- [ ] Documentation has been included (for new feature)

## Doc

Tell us where the doc can be found (docs folder, wiki, in the code...).

## Changes

Explain the changes that were made.
I added a new model for performance thresholds with fields: Indicator, timeline, fail_threshold, success_threshold, inclusive(to set the correct boundary points). 
I also added the model's API and the tests for the API

The idea is not to list exhaustively all the changes made (GitHub already provides a full diff), but to help the reviewers better understand:

- which specific file changes go together, e.g: when creating a table in the front-end, there usually is a config file that goes with it
- the reasoning behind some changes, e.g: deleted files because they are now redundant
- the behaviour to expect, e.g: tooltip has purple background color because the client likes it so, changed a key in the API response to be consistent with other endpoints

## How to test

Explain how to test your PR.
you need to run: `sudo docker compose exec iaso python manage.py test plugins.polio.tests.api.performance_thresholds`
to test the API

If a specific config is required explain it here: dataset, account, profile, etc.

## Print screen / video

Upload here print screens or videos showing the changes.

## Notes

Things that the reviewers should know:

- known bugs that are out of the scope of the PR
- other trade-offs that were made
- does the PR depends on a PR in [bluesquare-components](https://github.com/BLSQ/bluesquare-components)?
- should the PR be merged into another PR?

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: [POLIO-2034](https://bluesquare.atlassian.net/browse/POLIO-2034)
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.


[POLIO-2034]: https://bluesquare.atlassian.net/browse/POLIO-2034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[POLIO-2034]: https://bluesquare.atlassian.net/browse/POLIO-2034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Crebert08 committed:
Feat: Add Performance Dashboard Backend API

Crebert08 committed:
Fix: Eliminating the information exposure through an exception

Crebert08 committed:
Fix: Eliminating the unsorted and unformatted errors

Crebert08 committed:
Fix: Eliminating the single quotes found but double quotes preferred errors

Crebert08 committed:
Fix: Apply review comments on backend files and tests

Crebert08 committed:
Feat: Added the frontend to interact with the performance dashboard api

Crebert08 committed:
Fix: Fixing the pagination

Crebert08 committed:
Fix: Fixing the formats using ruff after review

Crebert08 committed:
Fix 2: Fixing the formats using ruff after review

Crebert08 committed:
Fix 3: Adding the missing translations after review

Crebert08 committed:
Fix 4: fixing the missing translations after review

Crebert08 committed:
Fix4: Changing the migrations

Crebert08 committed:
Fix 5: fixing the formatting in migrations after review

Crebert08 committed:
Fix 6: fixing the formating errors

Crebert08 committed:
Fix 7: Fixing the frontend translations

Crebert08 committed:
Fix 8: Adding the frontend translations

Crebert08 committed:
Fix: Fixing the code after the reviews

Crebert08 committed:
Fix:Auto-formatting

Crebert08 committed:
Fix:fixing a dependency issue

Crebert08 committed:
Fix: Changing the api hooks to call Vaccine not Antigen

Crebert08 committed:
Adding the model for threshold settings

Crebert08 committed:
Adding the model, API and tests for performance thresholds

Crebert08 committed:
Fix: adding the translations in french

Crebert08 committed:
Fix: adding missing translations

quang-le reviewed (changes requested):
Please change the base, so the changes from your previous PR don't appear in the diff and fill in the rest of the PR template (changes, screen capture if applicable etc).

If the PR is not ready to be reviewed, please put it in draft

quang-le committed:
Merge branch 'develop' into POLIO-2034-Performance-Dashboard-threshold-setting

quang-le committed:
POLIO-2034: refactor model and API

quang-le committed:
Merge branch 'develop' into POLIO-2034-Performance-Dashboard-threshold-setting

quang-le committed:
POLIO-2034: merge fixes

quang-le committed:
POLIO_2034: light refactor

- refactor "fake hook" into regular variable
- remove pop culture names from tests

quang-le committed:
POLIO-2034: refactor PerformanceThresholds API

- make API public for read-only for PerformanceThresholds and PerformanceDashboard
- Add AuditMixin to PerformanceThresholdsViewset for change logging
- Update tests
- Add convenience methods for json schema testing

quang-le committed:
POLIO-2034: Refactor PerformanceDashboard API and serializers

- remove created_by, updated_by fields from PerformanceDashboard model
- reflect change on API and tests
- add custom serializer for audit logs

quang-le committed:
POLIO-2034: add test for audit logs in perf dashboard

quang-le committed:
chore: fix ruff
